tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -203,11 +203,13 @@ Status ExecuteThunks(const std::string& module_name,
       [&] { return absl::StrCat(module_name, ":XLA GPU module"); },
       tsl::profiler::TraceMeLevel::kInfo);
 
+  ScopedAnnotation annotation(module_name);
+
   for (const std::unique_ptr<Thunk>& thunk : thunk_sequence) {
     // Annotate execution of this op if tracing was enabled when we started
     // running this module.  If tracing is enabled *while* we're running the
     // module, we won't get any data, but that's probably an OK trade-off.
-    ScopedAnnotation annotation([&] { return thunk->profile_annotation(); });
```

Comment:
this need to be reverted.
lambda is evaluated before construction of the string.
even thunk->profile_annotation() is pre-populated, however it retr